### PR TITLE
Proxy config: Add tls_certs_to_watch field

### PR DESCRIPTION
**Pull request by drichelson**
See comments [here](https://github.com/istio/istio/issues/11984) for context

### CODE REVIEW

File: mesh/v1alpha1/proxy.proto [hunk 1]
```diff
@@ -169,4 +169,7 @@ message ProxyConfig {
 
   // Tracing configuration to be used by the proxy.
   Tracing tracing = 19;
+
+  // Absolute paths to cert/key files to watch for changes (e.g. /etc/certs/key.pem,/etc/certs/cert.pem)
```
Comment (rshriram) [inline mesh/v1alpha1/proxy.proto]: Absolute path to certs/key files used for istio mTLS. If omitted, defaults to /etc/certs/...

File: mesh/v1alpha1/proxy.proto [hunk 2]
```diff
@@ -169,4 +169,8 @@ message ProxyConfig {
 
   // Tracing configuration to be used by the proxy.
   Tracing tracing = 19;
+
+  // Absolute path to certs/key files used for istio mTLS. If omitted, defaults to
+  // /etc/certs/cert-chain.pem,/etc/certs/key.pem,/etc/certs/root-cert.pem
+  repeated string tls_certs_to_watch = 21;
```
Comment (hzxuzhonghu) [inline mesh/v1alpha1/proxy.proto]: I have a question : how can you distinguish which is the key/cert file?

Comment (rshriram) [reply mesh/v1alpha1/proxy.proto]: we dont. pilot agent simply watches files in that path and if any of them change, it forces envoy to restart

**Pull request merged**